### PR TITLE
Cloudformation template to set up lambda that should run on merges with master

### PR DESCRIPTION
**Pull request by arunasank**
Set up a CloudFormation template to be run when we merge with master.

cc/ @ramyaragupathy

### CODE REVIEW

**Comment by arunasank** [comment]
**NOTE: We should create a `stats-collector-production` stack before merging this branch with `master`.**

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+version: 2.1
+jobs:
+  build:
+    working_directory: /home/circleci/app
+    docker:
+    - image: circleci/node:10
```
Comment (ramyaragupathy) [inline .circleci/config.yml]: Does this mean you use node v 10?

File: stats-collector.template.json [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                    "Fn::Sub": [
+                        "/aws/lambda/${name}",
+                        {
+                            "name": {
+                                "Fn::Sub": "${AWS::StackName}-MyLambda"
+                            }
+                        }
+                    ]
+                },
+                "RetentionInDays": 14
```
Comment (ramyaragupathy) [inline stats-collector.template.json]: What does this mean?

File: stats-collector.template.json [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                    "S3Key": "code.zip"
+                },
+                "Description": {
+                    "Fn::Sub": "MyLambda in the ${AWS::StackName} stack"
+                },
+                "FunctionName": {
+                    "Fn::Sub": "${AWS::StackName}-MyLambda"
+                },
+                "Handler": "index.handler",
+                "MemorySize": 128,
```
Comment (ramyaragupathy) [inline stats-collector.template.json]: Current memory requirement is close to 2gb

File: stats-collector.template.json [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                "Role": {
+                    "Fn::GetAtt": [
+                        "MyLambdaRole",
+                        "Arn"
+                    ]
+                },
+                "Runtime": "nodejs8.10",
+                "Timeout": 300
```
Comment (ramyaragupathy) [inline stats-collector.template.json]: Also entire lambda runs for about 12-13 mins

File: .circleci/config.yml (hunk 1 above)
Comment (arunasank) [reply .circleci/config.yml]: yes!

File: stats-collector.template.json (hunk 3 above)
Comment (arunasank) [reply stats-collector.template.json]: Memory or disk?

File: stats-collector.template.json (hunk 4 above)
Comment (arunasank) [reply stats-collector.template.json]: Will change this to 15 minutes. 👍

File: stats-collector.template.json (hunk 2 above)
Comment (arunasank) [reply stats-collector.template.json]: The logs are retained for 14 days, and then deleted.

File: stats-collector.template.json (hunk 3 above)
Comment (ramyaragupathy) [reply stats-collector.template.json]: memory requirement

**Pull request merged**